src/lines_to_params.rs

```diff
@@ -0,0 +1,165 @@
+        update_resources(source_actor_obj, source_resources);
+        update_resources(target_actor_obj, target_resources);
+
+        let ability_event = create_id_name_pair(py, "nari.types.ability", "Ability", ability);
+
+        let action_effect = PyModule::import(py, "nari.types.actioneffect").unwrap()
+            .getattr("ActionEffect").unwrap();
+
+        let action_effects = action_effects
+            .chunks(2)
+            .map(|x| {
+                let (param0, param1, severity, effect_category, value, flags, multiplier) = action_effect_from_params(x.to_vec());
+                let kwargs = PyDict::new(py);
+                kwargs.set_item("effect_category", effect_category).unwrap();
+                kwargs.set_item("severity", severity).unwrap();
+                kwargs.set_item("flags", flags).unwrap();
+                kwargs.set_item("value", value).unwrap();
+                kwargs.set_item("multiplier", multiplier).unwrap();
+                kwargs.set_item("additional_params", (param0, param1)).unwrap();
+                action_effect.call((), Some(kwargs)).unwrap()
+            }).collect::<Vec<&PyAny>>();
+
+        let kwargs = PyDict::new(py);
+        kwargs.set_item("timestamp", timestamp).unwrap();
+        kwargs.set_item("action_effects", action_effects).unwrap();
+        kwargs.set_item("source_actor", source_actor_obj).unwrap();
+        kwargs.set_item("target_actor", target_actor_obj).unwrap();
+        kwargs.set_item("ability", ability_event).unwrap();
+        kwargs.set_item("sequence_id", sequence).unwrap();
+
+        PyModule::import(py, "nari.types.event.ability").unwrap()
+            .getattr("Ability").unwrap().call((), Some(kwargs)).unwrap().to_object(py)
+    })
+}
+
+/// Params to action_effect
+#[pyfunction]
+#[pyo3(text_signature = "(params: list[str]) -> list[int]")]
+pub(crate) fn action_effect_from_params(inp: Vec<&str>) -> ActionEffectParams {
+    let mut num = parser::u32_from_param(inp[0]);
+    let param0 = (&num >> 24) as u8;
+    let param1 = (&num >> 16) as u8;
+    let severity = (&num >> 8) as u8;
+    let effect_category = num as u8;
+    num = parser::u32_from_param(inp[1]);
+    let value = (&num >> 16) as u16;
+    let flags = (&num >> 8) as u8;
+    let multiplier = num as u8;
+    (param0, param1, severity, effect_category, value, flags, multiplier)
+}
+
+/// Params to status_effect
+#[pyfunction]
+#[pyo3(text_signature = "(params: list[str]) -> (int, int, float, int")]
+pub(crate) fn status_effect_from_params(inp: Vec<&str>) -> StatusEffectParams {
+    let (param0, param1) = parser::u16x2_from_param(inp.get(0).unwrap());
+    (
+        param0,
+        param1,
+        parser::f32_from_param(inp.get(1).unwrap()),
+        parser::u32_from_param(inp.get(2).unwrap()),
+    )
+}
+
+/// Params to statuslist
+#[pyfunction]
+#[pyo3(text_signature = "(params: list[str]) -> list[any]")]
+pub(crate) fn statuslist_from_params(inp: Vec<&str>) -> StatuslistParams {
```

Comment:
Here too, but also why are we using `any` as a type? I thought it's largely banned.

Reply:
Look 2 above.

Reply:
Corrected to proper type. 
This function is currently not used in parsing the status list, as I didn't see the need to make that happen at this point, and will also get the same treatment as `ability_from_params` with a nicer export function instead of a giant tuple list typing.